### PR TITLE
Loki: Remove unused query editors

### PR DESCRIPTION
**Pull request by ivanahuckova**
**What this PR does / why we need it**:

This PR 
1. Removes Loki query editor that are not used anymore. Specifically: 
- public/app/plugins/datasource/loki/components/LokiExploreQueryEditor.tsx (previously used as editor in Explore)
- public/app/plugins/datasource/loki/components/LokiQueryEditor.tsx (previously used as editor in Dashboard and unknown app)

2. Renames `LokiQueryEditorSelector` to `LokiQueryEditor` as it is used ad Loki editor in all apps (besides cloud alerting, where we have a special editor)

3. Removes redundant reference to feature toggle `featureToggles.lokiQueryBuilder` that is not used anymore and was already removed. 

**Special notes for your reviewer**:

To test this PR - check that there are no visible changes between this branch and main
1. Run loki data source `make devenv sources=loki`
2. Try to run Loki query in dashboard, alerting, explore. Everywhere should be working query editor that looks the same. 

image
image
image

### CODE REVIEW

File: public/app/plugins/datasource/loki/components/LokiQueryEditorByApp.test.tsx [hunk 1]
```diff
@@ -40,24 +41,23 @@ describe('LokiQueryEditorByApp', () => {
...
+    expect(await waitFor(() => getByTestId(regularTestIds.editor))).toBeInTheDocument();
     expect(queryByTestId(alertingTestIds.editor)).toBeNull();
   });
 
-  it('should render regular query editor for dashboard', () => {
+  it('should render regular query editor for dashboard', async () => {
     const { getByTestId, queryByTestId } = setup(CoreApp.Dashboard);
 
-    expect(getByTestId(regularTestIds.editor)).toBeInTheDocument();
+    expect(await waitFor(() => getByTestId(regularTestIds.editor))).toBeInTheDocument();
```
Comment (matyax) [inline public/app/plugins/datasource/loki/components/LokiQueryEditorByApp.test.tsx:60]: You can consider:

```suggestion
    expect(await screen.findByTestId(regularTestIds.editor)).toBeInTheDocument();
```

https://testing-library.com/docs/example-findbytext/

File: public/app/plugins/datasource/loki/components/LokiQueryEditor.test.tsx [hunk 2]
```diff
@@ -1,72 +1,194 @@
...
-  const mockTemplateSrv: TemplateSrv = {
-    getVariables: jest.fn(),
-    replace: jest.fn(),
-    containsTemplate: jest.fn(),
-    updateTimeRange: jest.fn(),
-  };
-  const datasource = createLokiDatasource(mockTemplateSrv);
-  const onRunQuery = jest.fn();
-  const onChange = jest.fn();
+const datasource = new LokiDatasource(
```
Comment (matyax) [inline public/app/plugins/datasource/loki/components/LokiQueryEditor.test.tsx:39]: Why not using createLokiDatasource()?

File: public/app/plugins/datasource/loki/components/LokiQueryEditor.tsx [hunk 3]
```diff
@@ -1,71 +1,165 @@
...
-      data={data}
-      data-testid={testIds.editor}
-      range={range}
-      ExtraFieldElement={
-        <>
-          <LokiOptionFields
-            lineLimitValue={query?.maxLines?.toString() || ''}
-            resolution={query?.resolution || 1}
+    <>
+      <ConfirmModal
```
Comment (matyax) [inline public/app/plugins/datasource/loki/components/LokiQueryEditor.tsx:84]: If this is duplicated from `LokiQueryEditorSelector` I would seriously consider moving it to a component.

**Review by matyax — COMMENTED** [review]
Awesome! Let me know what you think of my comments.

File: public/app/plugins/datasource/loki/components/LokiQueryEditorByApp.test.tsx (hunk 1 above)
Comment (ivanahuckova) [reply public/app/plugins/datasource/loki/components/LokiQueryEditorByApp.test.tsx]: Yes I agree! It is also mentioned in one of my favourite and most useful blog posts https://kentcdodds.com/blog/common-mistakes-with-react-testing-library.

File: public/app/plugins/datasource/loki/components/LokiQueryEditor.test.tsx (hunk 2 above)
Comment (ivanahuckova) [reply public/app/plugins/datasource/loki/components/LokiQueryEditor.test.tsx:39]: I didn't change tests at all in this case. I just renames `LokiQueryEditorSelector` -> `LokiQueryEditor`, as previous `LokiQueryEditor` wasn't used anymore and I think `LokiQueryEditorSelector` should be named  `LokiQueryEditor` as that is what it is. I can (in the next PR) update the tests, but in this PR, I just wanted to remove unused things and rename and move `LokiQueryEditorSelector` where it belongs.

Comment (matyax) [reply public/app/plugins/datasource/loki/components/LokiQueryEditor.test.tsx:39]: Ohh I see, I see.

File: public/app/plugins/datasource/loki/components/LokiQueryEditor.test.tsx [hunk 4]
```diff
@@ -1,72 +1,194 @@
...
+  });
+
+  it('Can enable explain', async () => {
+    renderWithMode(QueryEditorMode.Builder);
+    expect(screen.queryByText(EXPLAIN_LABEL_FILTER_CONTENT)).not.toBeInTheDocument();
+    screen.getByLabelText('Explain').click();
+    expect(await screen.findByText(EXPLAIN_LABEL_FILTER_CONTENT)).toBeInTheDocument();
+  });
+
+  it('changes to code mode', async () => {
```
[the remark this answers is not in the document]
Comment (ivanahuckova) [reply public/app/plugins/datasource/loki/components/LokiQueryEditor.test.tsx:130]: good catch! Again, something I can tackle in the next PR, as here I wanted to change only essential tests that were failing.

File: public/app/plugins/datasource/loki/components/LokiQueryEditor.tsx (hunk 3 above)
Comment (ivanahuckova) [reply public/app/plugins/datasource/loki/components/LokiQueryEditor.tsx:84]: Same old, same old. 🙂 The renaming of file made this more confusing than it is. I should have splitted it into 2 PRs.

**Review by matyax — APPROVED** [review]

**Pull request merged**